fog/fog_timestamps/src/watcher.rs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2018-2021 The MobileCoin Foundation
+
+use mc_common::logger::{log, Logger};
+use mc_transaction_core::BlockIndex;
+use mc_watcher::watcher_db::WatcherDB;
+use mc_watcher_api::TimestampResultCode;
+use std::time::{Duration, Instant};
+
+/// Poll for new data every 10 ms
+const POLLING_FREQUENCY: Duration = Duration::from_millis(10);
+/// If a database invariant is violated, e.g. we get block but not block
+/// contents, it typically will not be fixed and so we won't be able to
+/// proceed. But bringing the server down is costly from ops POV because
+/// we will lose all the user rng's.
```

Comment:
user rngs do not apply here.

Reply:
please confirm that below is a better replacement for the comment

/// we will lose all the blocks

Reply:
We will not be losing any blocks. We have no way of knowing what effect broken database invariants has on the caller since we don't know what the caller is doing.
I would change `But bringing the server down is costly from ops POV because
/// we will lose all the user rng's.` to something like `Generally when an invariant is violated we would panic, but this code is used in services that are expensive to restart (such as the ingest enclave).`